Tests/SentryTests/SentryHubTests.swift

```diff
@@ -715,6 +715,45 @@ class SentryHubTests: XCTestCase {
         XCTAssertEqual(envelope, fixture.client.captureEnvelopeInvocations.first)
     }
 
+    func testCaptureEnvelope_WithUnhandledException() {
+          sut.startSession()
+
+          let beginSession = sut.session
+
+          let event = TestData.event
+          event.level = .error
+          event.exceptions = [TestData.exception]
+          event.exceptions?.first?.mechanism?.handled = false
+          sut.capture(SentryEnvelope(event: event))
+
+          let endSession = sut.session
+          XCTAssertNotEqual(beginSession, endSession)
+
+          //Check whether session was finished as crashed
+          let envelope = fixture.client.captureEnvelopeInvocations.first
+          let sessionEnvelopeItem = envelope?.items.first(where: { $0.header.type == "session" })
+
+          let json = (try! JSONSerialization.jsonObject(with: sessionEnvelopeItem!.data)) as! [String: Any]
+
+          XCTAssertNotNil(json["timestamp"])
```

Comment:
`m`: I think we should validate the timestamp instead of only checking if it's not nil.